mllib/src/main/scala/org/apache/spark/ml/r/RWrapperUtils.scala

```diff
@@ -41,5 +41,29 @@ object RWrapperUtils extends Logging {
         s"using new name $newFeaturesName instead")
       rFormula.setFeaturesCol(newFeaturesName)
     }
+
+    if (rFormula.getForceIndexLabel && data.schema.fieldNames.contains(rFormula.getLabelCol)) {
+      val newLabelName = s"${Identifiable.randomUID(rFormula.getLabelCol)}"
+      logWarning(s"data containing ${rFormula.getLabelCol} column and we force to index label, " +
+        s"using new name $newLabelName instead")
+      rFormula.setLabelCol(newLabelName)
+    }
+  }
+
+  /**
+   * Get the feature names and original labels from the schema
+   * of DataFrame transformed by RFormulaModel.
```

Comment:
It could have but not must, since it should be private. BTW, I found we should make object ```RWrapperUtils``` private[r], and will fix it soon.